api.planx.uk/editor/publish.ts

```diff
@@ -48,6 +49,18 @@ const publishFlow = async (
 
   try {
     const flattenedFlow = await dataMerged(req.params.flowId);
+
+    // Check the flattened flow to ensure that all sections (type 360) are still in valid positions (eg within `_root: { edges: [...]}`, not the edge of a portal)
+    //   If any are not, don't proceed with publishing and return a message to display in the Editor
+    const sectionTypeNodeIds = Object.entries(flattenedFlow).filter(([_nodeId, nodeData]) => nodeData?.type === 360).map(([nodeId, _nodeData]) => nodeId);
```

Comment:
Yep, good call! Now refactored to be two separate validation functions `sectionIsInFirstPosition` & `allSectionsOnRoot` which each return a boolean. These are still living in the `publish.ts` file since they're quite specific, not wider "helpers" etc. But easier to imagine how this might evolve into a growing list of validations this way :+1: